src/DotVVM.Framework/Controls/TextBox.cs

```diff
@@ -125,11 +125,21 @@ public static bool NeedsFormatting(IValueBinding binding) => binding != null &&
 
         protected internal override void OnPreRender(IDotvvmRequestContext context)
         {
-            isFormattingRequired = !string.IsNullOrEmpty(FormatString) ||
-                #pragma warning disable
+            if (!string.IsNullOrEmpty(FormatString) && (Type == TextBoxType.Number || Type == TextBoxType.Date))
```

Comment:
What about `TextBoxType.Time`? It also requires special format. Note that there is a few more allowed `type` value that `TextBoxType` does not support - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input, what about adding support for them?

Reply:
Yeah, we should add support for any input `types` where it make sense. 
But `TextBoxType.Time` type is little complicated. What C# type should user use? `TimeSpan` (currently not supported) or `DateTime` (what will happend if user bind the same property in `Date` and `Time` input type?)
We should make a **list of input types** where it make sense to threat them specially.